### ISSUE TITLE
Logging without `H.trace` in Python

### ISSUE THREAD

**Issue by jay-khatri** [highlight/highlight#5674]
**Describe the bug**
From Discord ([link](https://discord.com/channels/1026884757667188757/1090029472700432424/1120012209074995410)):
> i'm in a django project i've deployed to production. i'm seeing logs in highlight.io site, but only logs that are associated with an traced request from the frontend. if i make a request from the non-instrumented django admin site, or if i run django management commands on the backend, those logs will not appear (despite being logged to console/stdout with the new otel attributes). is this anticipated behavior, or have i done something wrong?

> it looks like if i wrap my code in an `H.trace` context, anything logged inside that context will go to highlight

**Expected behavior**

> example of a log line i'm seeing that i would expect to see in highlight
dotaleaguesite-api-1  | {"timestamp": "2023-06-18T04:02:47.781260Z", "level": "WARNING", "name": "django.request", "message": "Not Found: /favicon.ico", "otelSpanID": "666b1cc6f61c97ad", "otelTraceID": "d4ca98a374c869ae6defbf42939efc67", "otelTraceSampled": true, "otelServiceName": "", "status_code": 404, "request": "<WSGIRequest: GET '/favicon.ico'>"}

**Additional context**

**Comment by kmorrison** [comment]
Original reporter here
This code DOES NOT worked
```
import highlight_io

import logging
HIGHLIGHT_PROJECT_ID = ...
H = highlight_io.H(HIGHLIGHT_PROJECT_ID, instrument_logging=True)
logging.warning("hello from python")
```

This code DOES work
```
import highlight_io

import logging
H = highlight_io.H(HIGHLIGHT_PROJECT_ID, instrument_logging=True)
with H.trace("python"):
    logging.warning("hello from python")
```

Furthermore any request coming from my frontend (written in next, instrumented with highlight) gets info logs, but any request coming from non-instrumented frontend like django admin site or socket server does not show up in highlight.

**Comment by Vadman97** [comment]
Problematic code is https://github.com/highlight/highlight/blob/7166b2cfbf374760156af3d4e71fa7e0f0db8e90/sdk/highlight-py/highlight_io/sdk.py#L194-L197

we should create a span if we do not have a current span context.

**Comment by kmorrison** [comment]
Logging without a `H.trace()` appears to be working sometimes, but I've discovered something much weirder that was maybe the original problem all along. Only a certain percentage of my logs, (traced or untraced) appear in highlight dashboard. Weirder still, the ones that do appear won't all appear at once, but more will pop up between refreshes (even after all the testing code is complete).

![Screenshot 2023-06-29 at 11 43 20 PM](https://github.com/highlight/highlight/assets/652107/92abd075-9cd2-4bb5-9888-541e5a41e433)
![Screenshot 2023-06-29 at 11 43 36 PM](https://github.com/highlight/highlight/assets/652107/b8e2a2a0-08f1-428e-8dc4-3a934b61dcbd)

These two screenshots were taken a minute apart, we can see the log lines that I'd sent back-populating between shots.

Code I used to test was here 
```
import time
import highlight_io
import logging

H = highlight_io.H(********, instrument_logging=True)
with H.trace("backend"):
    for i in range(20):
        logging.info(f"hello from loop {i}")
        time.sleep(1)
```
I tried it with and without the trace, same effect
![Screenshot 2023-06-29 at 11 52 23 PM](https://github.com/highlight/highlight/assets/652107/49136061-fa7e-4305-8eca-5657a9a1a8ec)


```
import time
import highlight_io
import logging
for i in range(20):
    logging.info(f"hello from untraced loop {i}")
    time.sleep(1)
```

version
```
❯ pip freeze | grep highlight
highlight-io==0.5.2
```

Not sure if anyone is able to follow this, I'm happy to hop in a discord call and walk someone through my methodology, but for now it appears that backend python logging in highlight is nondeterministic.
